src/exp_grow.c

```diff
@@ -3,6 +3,12 @@
 
 void
 exp_grow_init(exp_grow_t *exp_grow) {
-	exp_grow->next = sz_psz2ind(HUGEPAGE);
+	/*
+	 * Enforce a minimal of 2M grow, which is convenient for the huge page
+	 * use cases.  Avoid using HUGEPAGE as the value though, because on some
+	 * platforms it can be very large (e.g. 512M on aarch64 w/ 64K pages).
+	 */
+	const size_t min_grow = (size_t)2 << 20;
+	exp_grow->next = sz_psz2ind(min_grow);
```

Comment:
As a check I applied this change (but not the others) and used the default --with-lg-hugepage value (29 on my ARM systems).  Unfortunately, with that configuration I still see VIRT reported at 10G whereas when I set --with-lg-hugepage=21 (same as intel) I get an as-expected 1.3G VIRT, in my test.

Maybe there is someplace else that needs to be addressed, as well?

Reply:
@madscientist good catch! Yes I missed the case in the base allocator which has a similar HUGEPAGE value hardcoded. See the changes in `base.c` to remedy that as well. The first commit in this PR should fix the issue properly now.

Reply:
I've tried this version and it does seem to solve the problem!  I will be away from my system for a week (going diving!) so I won't be able to test further until I get back.  Thanks for the quick update and great help!

Reply:
@madscientist Thank you again and have a great trip!